components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
so, what is the conclusion for now? Just the old API was misleading, and if we have to make a change anyway, I would vote to fix the API _iff_ it is the same amount of work.

I agree about being careful going forward, just if you have to break things for a serious reason, I think it makes sense to fix/improve all the related things (if any) if it is not much more work